server/src/main/java/org/elasticsearch/action/ingest/SimulatePipelineResponse.java

```diff
@@ -19,22 +19,93 @@
 
 package org.elasticsearch.action.ingest;
 
+import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.common.ParseField;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
 import org.elasticsearch.common.xcontent.ToXContentObject;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentParser.Token;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
 public class SimulatePipelineResponse extends ActionResponse implements ToXContentObject {
     private String pipelineId;
     private boolean verbose;
     private List<SimulateDocumentResult> results;
 
+    @SuppressWarnings("unchecked")
+    public static final ConstructingObjectParser<SimulatePipelineResponse, Void> PARSER =
+        new ConstructingObjectParser<>(
+            "simulate_pipeline_response",
+            true,
+            a -> {
+                List<SimulateDocumentResult> results = (List<SimulateDocumentResult>)a[0];
+                boolean verbose = false;
+                if (results.size() > 0) {
+                    if (results.get(0) instanceof SimulateDocumentVerboseResult) {
+                        verbose = true;
+                    }
+                }
+                return new SimulatePipelineResponse(null, verbose, results);
+            }
+        );
+    static {
+        PARSER.declareObjectArray(
+            constructorArg(),
+            (parser, context) -> {
+                Token token = parser.currentToken();
+                ensureExpectedToken(Token.START_OBJECT, token, parser::getTokenLocation);
+                SimulateDocumentResult result = null;
+                while ((token = parser.nextToken()) != Token.END_OBJECT) {
+                    ensureExpectedToken(token, Token.FIELD_NAME, parser::getTokenLocation);
+                    String fieldName = parser.currentName();
+                    token = parser.nextToken();
+                    if (token == Token.START_ARRAY) {
+                        switch (fieldName) {
+                            case SimulateDocumentVerboseResult.PROCESSOR_RESULT_FIELD:
```

Comment:
can you do else if here please?

Reply:
You mean something like:
```
if ( fieldName.equals(PROCESSOR_RESULT_FIELD)) { /* do something */ }
else { /* skip children */ }
```
I wanted to make it similar to the one below so I did a switch statement. But I agree it would make more sense with an if.

Reply:
yes that is what I meant, thank you!